From ceabb60844ff6bb220cf05f040031c51d4a69d9d Mon Sep 17 00:00:00 2001
From: thierry tournie <ttournie@thierrys-MacBook-Pro.local>
Date: Thu, 23 Jun 2016 10:19:08 +0200
Subject: [PATCH 1/1] kostym patch

---
 includes/common.inc | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/includes/common.inc b/includes/common.inc
index 7d79841..5d2ec29 100644
--- a/includes/common.inc
+++ b/includes/common.inc
@@ -5513,6 +5513,24 @@ function drupal_system_listing($mask, $directory, $key = 'name', $min_depth = 1)
     $searchdir[] = "$config/$directory";
   }

+  // To allow component modules to be discovered in the theme
+  if($directory == 'modules'){
+    if(!function_exists('kostym_list_themes')){
+      module_load_include('module', 'kostym');
+    }
+
+    // If we are in the middle of an install or update scan all themes
+    $only_enabled_themes = !defined('MAINTENANCE_MODE');
+
+    foreach(kostym_list_themes($only_enabled_themes) as $theme_info){
+      $component_path = $theme_info->path . '/kostym_components';
+
+      if(is_dir($component_path)){
+        $searchdir[] = $component_path;
+      }
+    }
+  }
+
   // By default, do not check for files in common special-purpose directories.
   // The folders here are front-end related and they have been added to avoid
   // issues with Drupal recursive scanning. In this case, we added node_modules
@@ -5555,6 +5573,15 @@ function drupal_system_listing($mask, $directory, $key = 'name', $min_depth = 1)
     $files = array_merge($files, $files_to_add);
   }

+  // Avoid components to be added as themes
+  if($directory == 'themes'){
+    foreach($files as $file_key => $file){
+      if(preg_match('/kostym_components\/' . $file_key . '\/' . $file_key . '\.info$/', $file->uri)){
+        unset($files[$file_key]);
+      }
+    }
+  }
+
   return $files;
 }

--
2.6.4
