diff --git a/includes/common.inc b/includes/common.inc
index 0a91cf2..4f7af43 100644
--- a/includes/common.inc
+++ b/includes/common.inc
@@ -5514,6 +5514,19 @@ function drupal_system_listing($mask, $directory, $key = 'name', $min_depth = 1)
     $searchdir[] = "$config/$directory";
   }
 
+  // To allow component modules to be discovered in the theme
+  if($directory == 'modules'){
+    foreach(list_themes() as $theme_info){
+      if($theme_info->status){
+        $component_path = drupal_get_path('theme', $theme_info->name) . '/kostym_components';
+
+        if(is_dir($component_path)){
+          $searchdir[] = $component_path;
+        }
+      }
+    }
+  }
+
   // By default, do not check for files in common special-purpose directories.
   // The folders here are front-end related and they have been added to avoid
   // issues with Drupal recursive scanning. In this case, we added node_modules
@@ -5556,6 +5569,15 @@ function drupal_system_listing($mask, $directory, $key = 'name', $min_depth = 1)
     $files = array_merge($files, $files_to_add);
   }
 
+  // Avoid components to be added as themes
+  if($directory == 'themes'){
+    foreach($files as $file_key => $file){
+      if(preg_match('/kostym_components\/' . $file_key . '\/' . $file_key . '\.info$/', $file->uri)){
+        unset($files[$file_key]);
+      }
+    }
+  }
+
   return $files;
 }
 
